lib/uplinkc/object.go

```diff
@@ -95,11 +96,11 @@ type Upload struct {
 
 // upload uploads a new object, if authorized.
 //export upload
-func upload(cBucket C.BucketRef, path *C.char, cOpts *C.UploadOptions, cErr **C.char) (downloader C.UploaderRef) {
+func upload(cBucket C.BucketRef, path *C.char, cOpts *C.UploadOptions, cErr **C.char) C.UploaderRef {
```

Comment:
Does this apply here?  "Cgo translates C types into equivalent unexported Go types. Because the translations are unexported, a Go package should not expose C types in its exported API: a C type used in one Go package is different from the same C type used in another. "

Reply:
This change is just a superficial one. The unused, named return argument was named the wrong thing; hence, just un-naming it. 

That line from the cgo is referring to a situation like this: 
```
// uplink_definitions.h
typedef struct Bucket { long _handle; } BucketRef; 
```
```
package one
// #include "uplink_definitions.h"
import "C"
type BucketRef =  C.BucketRef
func AssertEqualZeroValue() {
    assert(BucketRef{} == C.BucketRef{})
}
```
```
package two
// #include "uplink_definitions.h"
import "C"
type BucketRef =  C.BucketRef
func AssertEqualZeroValue() {
    assert(BucketRef{} == C.BucketRef{})
}
```
```
package main

import "reflect"
import "one"
import "two"

assert(one.BucketRef{} == two.BucketRef{}) // should fail
assert(C.BucketRef{} == one.BucketRef{}) // should fail
assert(C.BucketRef{} == two.BucketRef{}) // should fail
```
**Disclaimer**: I haven't run any of this code :/

Here's a related SO question: [Why are Go C types different when they come from different modules?](https://stackoverflow.com/questions/47117064/why-are-go-c-types-different-when-they-come-from-different-modules) and here's a github issue discussing changing this: [cmd/cgo: make identical C types identical Go types across packages](https://github.com/golang/go/issues/13467).